### PR TITLE
Improve documentation on contributing to documentation

### PR DESCRIPTION
**Pull request by cocobennett**
Fixes https://github.com/readthedocs/readthedocs.org/issues/8042

### CODE REVIEW

**Comment by astrojuanlu** [comment]
Hi @cocobennett , thanks for your interest in contributing to Read the Docs! 😊 

I notice that this PR is still marked as draft, so let us know when it's ready to receive feedback.

Also, I noticed that the branch contains a spurious merge commit, I think it would be better if you `rebase` on top of `readthedocs:master` and force push, to keep the history as clean as possible.

**Comment by cocobennett** [comment]
> Hi @cocobennett , thanks for your interest in contributing to Read the Docs! 😊
> 
> I notice that this PR is still marked as draft, so let us know when it's ready to receive feedback.
> 
> Also, I noticed that the branch contains a spurious merge commit, I think it would be better if you `rebase` on top of `readthedocs:master` and force push, to keep the history as clean as possible.

Thanks @astrojuanlu, you're welcome and I'm happy to help! Right now, I'm trying to figure out how to contribute locally so I can write up those instructions and will let you know!

And great tip, thank you. I will rebase instead

**Comment by cocobennett** [comment]
@astrojuanlu Actually, if you could help me, I am having one issue. I've followed the instructions here https://docs.readthedocs.io/en/stable/development/install.html (including the read-only settings) 
image
but when I navigate to http://community.dev.readthedocs.io/, it looks like this where it's not formatted correctly. Have you seen this before?
image

**Comment by stsewd** [comment]
> @astrojuanlu Actually, if you could help me, I am having one issue. I've followed the instructions here https://docs.readthedocs.io/en/stable/development/install.html (including the read-only settings)
> image
> but when I navigate to http://community.dev.readthedocs.io/, it looks like this where it's not formatted correctly. Have you seen this before?
> image

when editing the policy you need to leave the "prefix" field empty and click on add. After your first build you'll see that there is another bucket named "media" and you'll need to do the same. We should make this more clear with a screenshot maybe

**Comment by cocobennett** [comment]
> > @astrojuanlu Actually, if you could help me, I am having one issue. I've followed the instructions here https://docs.readthedocs.io/en/stable/development/install.html (including the read-only settings)
> 
> when editing the policy you need to leave the "prefix" field empty and click on add. After your first build you'll see that there is another bucket named "media" and you'll need to do the same. We should make this more clear with a screenshot maybe

Thank you, that worked! Now I'm one step closer. Not sure if the problem was me or the documentation, probably me :D
image

**Comment by cocobennett** [comment]
Moving this to Ready for Review because I've added basic instructions for both in the UI and locally, as well as rebased correctly. I need help in step 2 of the local instructions as I myself don't know how to get from http://community.dev.readthedocs.io/ to my development version of the documentation. Thanks!

**Comment by cocobennett** [comment]
Hey there, I'd really like to help contribute here for myself and other newcomers. I'm trying to figure out how to find my development version of the docs after getting `http://community.dev.readthedocs.io/` up and running.

I've tried clicking the "Documentation" link at the very bottom but this leads to me the actual live documentation
image

I've also tried guessing various urls based on the url generated from this PR itself, which is `https://docs--8082.org.readthedocs.build/en/8082/`. I've guessed the following urls without success to find my locally hosted documentation:

* http://community.docs.readthedocs.io/
* http://community.dev.readthedocs.io/en/stable/docs
* http://docs.org.readthedocs.build/

Any tips would be appreciated.

**Comment by astrojuanlu** [comment]
By the way, I opened https://github.com/readthedocs/readthedocs.org/issues/8090 to clarify the MinIO settings, I agree they're confusing.

**Comment by astrojuanlu** [comment]
> I've tried clicking the "Documentation" link at the very bottom but this leads to me the actual live documentation

I wonder if the links on the footer should be relative instead of absolute.

> I'm trying to figure out how to find my development version of the docs after getting http://community.dev.readthedocs.io/ up and running.

You'd need to follow some extra steps that, to my knowledge, are not documented yet:

1. Log in as `admin/admin`
2. Go to the "Read the Docs" project
3. Build `latest` documentation
4. Wait for it to finish and open http://read-the-docs.community.dev.readthedocs.io/en/latest/

In my opinion, these steps should be in https://docs.readthedocs.io/en/stable/development/install.html#set-up-your-environment, as a way to do an end-to-end test to check if everything worked. cc @readthedocs/core

**Comment by humitos** [comment]
> In my opinion, these steps should be in [docs.readthedocs.io/en/stable/development/install.html#set-up-your-environment](https://docs.readthedocs.io/en/stable/development/install.html#set-up-your-environment), as a way to do an end-to-end test to check if everything worked. cc @readthedocs/core

Sounds good to me! I'd add a section after "Set up your environment" communicating this and making sure that everything is working as expected.

File: docs/development/docs.rst [hunk 1]
```diff
@@ -10,27 +10,48 @@ The docs are kept in the ``docs/`` directory at the top of the source tree.
...
+3. navigate to the section of the documentation you worked on to verify your changes
+
+Contributing from your local machine
+------------------------------------
+
+If you're making large changes to the documentation, you'll want to verify those changes locally before pushing upstream.
+
+1. follow the instructions to `Set up your environment <https://docs.readthedocs.io/en/stable/development/install.html#set-up-your-environment>`__
+2. navigate to ??? to view your development version of the Read The Docs documentation
+3. as long as all your containers are up as shown in step 6 of `Set up your environment <https://docs.readthedocs.io/en/stable/development/install.html#set-up-your-environment>`__, your changes should be reflected automatically on each save
```
Comment (astrojuanlu) [inline docs/development/docs.rst:29]: I think the "as long as..." clarification is perhaps a bit redundant.

Also, "changes should be reflected automatically on each save" might be misleading: builds don't automatically trigger on disk changes, nor on new local commits (just tested it). Therefore, we should expand these instructions a bit and explain that we would need to rebuild the docs for every change.

**Review by astrojuanlu — COMMENTED** [review]
Left a couple of comments, but I think this is a step in the right direction!

**Comment by cocobennett** [comment]
Thanks for the review and for the instructions @astrojuanlu ! I'm not sure which steps should be in "Setup your Environment" vs what steps should be listed in the file I'm editing here, so I've added them all to the current file for now. Maybe just the first step to ensure logging in as admin/admin works?

I ran into another issue though, both this link (http://read-the-docs.community.dev.readthedocs.io/en/latest/) and the "View Docs" button in the UI both lead me to this page instead of the documentation
image

File: docs/development/docs.rst [hunk 2]
```diff
@@ -10,27 +10,50 @@ The docs are kept in the ``docs/`` directory at the top of the source tree.
...
+1. click the checkmark next to your commit and it will expand to have multiple options
+2. right-click the "details" link next to the "docs/readthedocs.org:docs" item
+3. navigate to the section of the documentation you worked on to verify your changes
+
+Contributing from your local machine
+------------------------------------
+
+If you're making large changes to the documentation, you'll want to verify those changes locally before pushing upstream.
+
+1. follow the instructions to `Set up your environment <https://docs.readthedocs.io/en/stable/development/install.html#set-up-your-environment>`__
```
Comment (ericholscher) [inline docs/development/docs.rst]: I don't think we actually want users to _install Read the Docs_ to contribute docs. People should be able to just have a Sphinx environment setup locally and build the docs through that. 

As you've seen in this PR, having a full RTD install is _a ton of work_ and requires understanding a great depth of technical stuff. I think we should strongly suggest folks just do work in git and let the PR builds do the building, and alternatively to just get a local Sphinx install.

Comment (cocobennett) [reply docs/development/docs.rst]: Ah, I see what you're saying. Agreed, that would be much easier. I will go through this path myself then update the instructions here.

Comment (ericholscher) [reply docs/development/docs.rst]: @cocobennett Sounds good. Having a local install is great (and you've already found lots of bugs :D) but keeping this barrier to entry low is definitely going to be a benefit. Thanks again for your help on this!

**Comment by ericholscher** [comment]
> I ran into another issue though, both this link (http://read-the-docs.community.dev.readthedocs.io/en/latest/) and the "View Docs" button in the UI both lead me to this page instead of the documentation

I believe this is another Minio permissions issue. You need to do the same thing for the `media` bucket as you did for the `static` bucket: https://github.com/readthedocs/readthedocs.org/pull/8082#issuecomment-816875657

File: docs/development/docs.rst [hunk 3]
```diff
@@ -10,27 +10,50 @@ The docs are kept in the ``docs/`` directory at the top of the source tree.
...
+1. click the checkmark next to your commit and it will expand to have multiple options
+2. right-click the "details" link next to the "docs/readthedocs.org:docs" item
+3. navigate to the section of the documentation you worked on to verify your changes
+
+Contributing from your local machine
+------------------------------------
+
+If you're making large changes to the documentation, you'll want to verify those changes locally before pushing upstream.
+
+1. follow the instructions to `Set up your environment <https://docs.readthedocs.io/en/stable/development/install.html#set-up-your-environment>`_
```
Comment (stsewd) [inline docs/development/docs.rst:27]: When linking to internal pages we use the `:ref:` or `:doc:` roles https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#cross-referencing-arbitrary-locations. You can see an example here https://github.com/readthedocs/readthedocs.org/blob/8009046a9c416816ddc3448c8427d9c683f0e65d/docs/config-file/v2.rst#L259-L262

But looks like you are going to change this anyway :D

File: docs/development/docs.rst [hunk 4]
```diff
@@ -10,27 +10,50 @@ The docs are kept in the ``docs/`` directory at the top of the source tree.
    However, if there is going to be periodic/bigger contributions, it may be a
    good idea to install the Sphinx requirements to build our docs.
 
+Contributing through the Github UI
+----------------------------------
+
+If you're making small changes to the documentation, you can verify those changes through the documentation generated when you open a PR and can be accessed using the Github UI.
```
Comment (stsewd) [inline docs/development/docs.rst:16]: For long lines, you can break them after the comma https://docs.readthedocs.io/en/stable/development/docs.html#content

**Comment by cocobennett** [comment]
Done for the night. Hit a roadblock with a bunch of pip installs not found in the `requirements/docs.txt`. Not sure if it's a path issue on my end or something else. Will check again tomorrow

**Comment by astrojuanlu** [comment]
Indeed, `requirements/docs.txt` does not contain all the required dependencies. The command `tox -e docs`, which is very superficially mentioned in https://docs.readthedocs.io/en/stable/development/tests.html?highlight=docs#testing, works<sup>1</sup>. Perhaps we should just bring it up in the contribution documentation as well?

<details>
This is essentially what `tox -e docs` does:

```
(.venv37) $ pip install -r requirements/testing.txt
(.venv37) $ pip install -r requirements/docs.txt
(.venv37) $ cd docs && make html 
```

The `docs.txt` is mentioned in the tox configuration with a `TODO`:

https://github.com/readthedocs/readthedocs.org/blob/db24ee1fff270717ec359a74f9a330fa78912829/tox.ini#L31

and there was some previous discussion about what would be the best way to do this: https://github.com/readthedocs/readthedocs.org/pull/7037#issuecomment-644960726

It would be interesting to analyze if we are already in a position to simplify this.
</details>

<sup>1 it's easier if run from Python 3.7, because `orjson` is pinned to `2.0.7`, which doesn't have wheels for >= 3.8 and requires the Rust toolchain to compile</sup>

**Comment by cocobennett** [comment]
> Indeed, `requirements/docs.txt` does not contain all the required dependencies. The command `tox -e docs`, which is very superficially mentioned in https://docs.readthedocs.io/en/stable/development/tests.html?highlight=docs#testing, works1. Perhaps we should just bring it up in the contribution documentation as well?
> 
> 1 it's easier if run from Python 3.7, because `orjson` is pinned to `2.0.7`, which doesn't have wheels for >= 3.8 and requires the Rust toolchain to compile

Thanks! `tox -e docs` seems to get me past that step! But sorry guys, I'm running into another issue. Posting it here because I'll be away from the computer for a few hours.

In a clean venv, I ran `tox -e docs` and ran into an error about needing rust, so I followed the instructions given in the error code to this [url](https://rustup.rs/). After that, I'm still running into errors. I tried [this](https://github.com/rust-lang/cargo/issues/3215) and [this](https://zoomadmin.com/HowToInstall/UbuntuPackage/pkg-config) and still getting the following error and not sure how to proceed at the moment.

```
Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: /home/coco/Projects/readthedocs.org/.tox/docs/bin/python /home/coco/Projects/readthedocs.org/.tox/docs/lib/python3.8/site-packages/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmps417n3va
         cwd: /tmp/pip-install-xsa2bpx3/orjson
    Complete output (29 lines):
    💥 pyo3-pack failed
      Caused by: Cargo metadata failed: Error during execution of `cargo metadata`:     Updating crates.io index
        Updating git repository `https://github.com/PyO3/pyo3.git`
        Updating git repository `https://github.com/ijl/json.git`
    error: failed to get `serde_json` as a dependency of package `orjson v2.0.7 (/tmp/pip-install-xsa2bpx3/orjson)`

    Caused by:
      failed to load source for dependency `serde_json`

    Caused by:
      Unable to update https://github.com/ijl/json.git?rev=2c0c55b628c9177543283a1535549e1dcb6a870a

    Caused by:
      failed to fetch into: /home/coco/.cargo/git/db/json-240b451de80eb693

    Caused by:
      failed to authenticate when downloading repository

      * attempted to find username/password via git's `credential.helper` support, but failed

      if the git CLI succeeds then `net.git-fetch-with-cli` may help here
      https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli

    Caused by:
      failed to acquire username/password from local configuration

    Checking for Rust toolchain....
    Running `pyo3-pack pep517 write-dist-info --metadata-directory /tmp/pip-modern-metadata-6xla9il6 --manylinux=off --rustc-extra-args=-C target-feature=+sse2 --strip=on`
    Error: Command '['pyo3-pack', 'pep517', 'write-dist-info', '--metadata-directory', '/tmp/pip-modern-metadata-6xla9il6', '--manylinux=off', '--rustc-extra-args=-C target-feature=+sse2', '--strip=on']' returned non-zero exit status 1.
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/coco/Projects/readthedocs.org/.tox/docs/bin/python /home/coco/Projects/readthedocs.org/.tox/docs/lib/python3.8/site-packages/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmps417n3va Check the logs for full command output.

=========================================================================== log end ===========================================================================ERROR: could not install deps [-r/home/coco/Projects/readthedocs.org/requirements/testing.txt]; v = InvocationError('/home/coco/Projects/readthedocs.org/.tox/docs/bin/python -m pip install -r/home/coco/Projects/readthedocs.org/requirements/testing.txt', 1)
___________________________________________________________________________ summary ___________________________________________________________________________ERROR:   docs: could not install deps [-r/home/coco/Projects/readthedocs.org/requirements/testing.txt]; v = InvocationError('/home/coco/Projects/readthedocs.org/.tox/docs/bin/python -m pip install -r/home/coco/Projects/readthedocs.org/requirements/testing.txt', 1)
```

**Comment by astrojuanlu** [comment]
Thanks for sticking through! It turns out this was a weak spot in our documentation that deserved some attention 🙏🏽

**Comment by humitos** [comment]
> it's easier if run from Python 3.7, because `orjson` is pinned to `2.0.7`

Just in case, we are using Python 3.6.x in all our stack. So, I'd recommend to use that version that's well tested.

**Comment by cocobennett** [comment]
> > it's easier if run from Python 3.7, because `orjson` is pinned to `2.0.7`
> 
> Just in case, we are using Python 3.6.x in all our stack. So, I'd recommend to use that version that's well tested.

Even with a python3.6 environment, I ran `tox -e docs` then get an error to install [rust](https://rustup.rs/), then I'm still seeing more errors.

```
Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: /home/coco/Projects/readthedocs.org/.tox/docs/bin/python /home/coco/Projects/readthedocs.org/.tox/docs/lib/python3.8/site-packages/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpiqmtppp0
         cwd: /tmp/pip-install-a4ti01ib/orjson
    Complete output (24 lines):
    💥 pyo3-pack failed
      Caused by: Cargo metadata failed: Error during execution of `cargo metadata`:     Updating crates.io index
    error: failed to get `encoding_rs` as a dependency of package `orjson v2.0.7 (/tmp/pip-install-a4ti01ib/orjson)`

    Caused by:
      failed to load source for dependency `encoding_rs`

    Caused by:
      Unable to update registry `https://github.com/rust-lang/crates.io-index`

    Caused by:
      failed to fetch `https://github.com/rust-lang/crates.io-index`

    Caused by:
      network failure seems to have happened
      if a proxy or similar is necessary `net.git-fetch-with-cli` may help here
      https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli

    Caused by:
      SSL error: error:1408F119:SSL routines:ssl3_get_record:decryption failed or bad record mac; class=Ssl (16)

    Checking for Rust toolchain....
    Running `pyo3-pack pep517 write-dist-info --metadata-directory /tmp/pip-modern-metadata-npry5xvm --manylinux=off --rustc-extra-args=-C target-feature=+sse2 --strip=on`
    Error: Command '['pyo3-pack', 'pep517', 'write-dist-info', '--metadata-directory', '/tmp/pip-modern-metadata-npry5xvm', '--manylinux=off', '--rustc-extra-args=-C target-feature=+sse2', '--strip=on']' returned non-zero exit status 1.
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/coco/Projects/readthedocs.org/.tox/docs/bin/python /home/coco/Projects/readthedocs.org/.tox/docs/lib/python3.8/site-packages/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpiqmtppp0 Check the logs for full command output.

=========================================================================== log end ===========================================================================ERROR: could not install deps [-r/home/coco/Projects/readthedocs.org/requirements/testing.txt]; v = InvocationError('/home/coco/Projects/readthedocs.org/.tox/docs/bin/python -m pip install -r/home/coco/Projects/readthedocs.org/requirements/testing.txt', 1)
___________________________________________________________________________ summary ___________________________________________________________________________ERROR:   docs: could not install deps [-r/home/coco/Projects/readthedocs.org/requirements/testing.txt]; v = InvocationError('/home/coco/Projects/readthedocs.org/.tox/docs/bin/python -m pip install -r/home/coco/Projects/readthedocs.org/requirements/testing.txt', 1)
```

Maybe using the UI should be good enough if contributors don't setup the full stack?

**Comment by astrojuanlu** [comment]
Beware! For some reason, I think your `tox -e docs` is using a Python 3.8 interpreter (see for example the path `/home/coco/Projects/readthedocs.org/.tox/docs/lib/python3.8/site-packages/pep517/_in_process.py`

**Comment by cocobennett** [comment]
Thanks for the hint @astrojuanlu , that was the issue!!! I ran the pip commands separately since tox was using 3.8 and that worked for me :) 
image

I'm ready for the final review now. Thanks for all the fast feedback!

File: requirements/testing.txt [hunk 5]
```diff
@@ -17,7 +17,7 @@ execnet==1.8.0
 # supported platform.
 # mercurial-scm.org/wiki/SupportedPythonVersions
 # (Pinned to 4.4.2 since what we need for testing is still useful)
-Mercurial==4.4.2  # pyup: ignore
+# Mercurial==4.4.2  # pyup: ignore
```
Comment (cocobennett) [inline requirements/testing.txt:20]: For some reason, this kept failing, but everything else worked when I commented it out. I'm sure this is used elsewhere so I know this should be handled a better way

File: docs/development/docs.rst [hunk 6]
```diff
@@ -10,27 +10,84 @@ The docs are kept in the ``docs/`` directory at the top of the source tree.
...
+
+   Run the `tox` command below to install the requirements.
+   More information about the `docs` option can be found on :ref:`development tests <development/tests:Testing>` section.
+
+   .. prompt:: bash
+
+      cd readthedocs.org
+      tox -e docs
+
+   Alternatively, you can try running the following commands if the `tox` command does not work for you.
```
Comment (cocobennett) [inline docs/development/docs.rst:49]: After fixing my local setup and the tests, this step isn't needed. Do you think others will run into this or should I remove it?

File: docs/development/docs.rst [hunk 7]
```diff
@@ -10,27 +10,84 @@ The docs are kept in the ``docs/`` directory at the top of the source tree.
...
    good idea to install the Sphinx requirements to build our docs.
 
+Contributing through the Github UI
+----------------------------------
+
+If you're making small changes to the documentation,
+you can verify those changes through the documentation generated when you open a PR and can be accessed using the Github UI.
+
+#. click the checkmark next to your commit and it will expand to have multiple options
+#. right-click the "details" link next to the "docs/readthedocs.org:docs" item
```
Comment (stsewd) [inline docs/development/docs.rst:20]: I think an image could be included below this list

Screenshot_2021-04-14 Improve documentation on contributing to documentation by cocobennett · Pull Request #8082 · readthed

File: docs/development/docs.rst [hunk 8]
```diff
@@ -10,27 +10,84 @@ The docs are kept in the ``docs/`` directory at the top of the source tree.
    However, if there is going to be periodic/bigger contributions, it may be a
    good idea to install the Sphinx requirements to build our docs.
```
Comment (stsewd) [inline docs/development/docs.rst:11]: we can remove this comment now

File: docs/development/docs.rst [hunk 9]
```diff
@@ -10,27 +10,84 @@ The docs are kept in the ``docs/`` directory at the top of the source tree.
...
+#. click the checkmark next to your commit and it will expand to have multiple options
+#. right-click the "details" link next to the "docs/readthedocs.org:docs" item
+#. navigate to the section of the documentation you worked on to verify your changes
+
+Contributing from your local machine
+------------------------------------
+
+.. note::
+
+    As of April 2021, you must use python 3.7 or lower to generate the documentation.
```
Comment (stsewd) [inline docs/development/docs.rst:28]: think you were able to build fine with 3.6

File: docs/development/docs.rst (hunk 6 above)
Comment (stsewd) [reply docs/development/docs.rst:49]: yeah, there is no need to run tox here

File: docs/development/docs.rst [hunk 10]
```diff
@@ -10,27 +10,84 @@ The docs are kept in the ``docs/`` directory at the top of the source tree.
...
+
+#. build the documents
+
+   .. prompt:: bash
+
+      cd docs
+      make html
+
+#. the documents will be available in the `_build` folder.
+   Double-click the `index.html` file to open the documentation in a webpage
```
Comment (stsewd) [inline docs/development/docs.rst:65]: I think we can make it more easy by installing https://pypi.org/project/sphinx-autobuild/ in our requirements/docs.txt, the users can run `make livehtml` and open 127.0.0.1:4444/ in their browser, the page will refresh on each change

**Review by stsewd — COMMENTED** [review]
thanks! I left a suggestion to use sphinx-autobuild

File: docs/development/docs.rst (hunk 10 above)
Comment (cocobennett) [reply docs/development/docs.rst:65]: Genius!

File: docs/development/docs.rst [hunk 11]
```diff
@@ -5,32 +5,77 @@ As one might expect,
...
+   .. image:: /img/details_link.png
+
+#. navigate to the section of the documentation you worked on to verify your changes
+
+Contributing from your local machine
+------------------------------------
+
+.. note::
+
+    As of April 2021, you must use python 3.6 to generate the documentation.
```
Comment (stsewd) [inline docs/development/docs.rst:27]: I think we can remove the date from this warning, we can update this note when we update our code base and tests to support another version of python (3.9)

File: docs/img/details-link.png:Zone.Identifier [hunk 12]
```diff
@@ -0,0 +1,4 @@
+[ZoneTransfer]
+ZoneId=3
+ReferrerUrl=https://user-images.githubusercontent.com/4975310/114787277-268cd480-9d45-11eb-9c2f-cb5066d34045.png
+HostUrl=https://user-images.githubusercontent.com/4975310/114787277-268cd480-9d45-11eb-9c2f-cb5066d34045.png
```
Comment (stsewd) [inline docs/img/details-link.png:Zone.Identifier:4]: Looks like this file isn't needed :)

**Review by stsewd — APPROVED** [review]
Thanks for sticking with us! Sorry you had to setup the whole RTD instance with docker when it wasn't needed.

**Comment by cocobennett** [comment]
> Thanks for sticking with us! Sorry you had to setup the whole RTD instance with docker when it wasn't needed.

No problem! I learned a lot. So happy to see this through :)

File: docs/development/docs.rst [hunk 13]
```diff
@@ -10,27 +10,48 @@ The docs are kept in the ``docs/`` directory at the top of the source tree.
    However, if there is going to be periodic/bigger contributions, it may be a
    good idea to install the Sphinx requirements to build our docs.
 
+Contributing through the Github UI
+----------------------------------
+
+If you're making small changes to the documentation, you can verify those changes through the documentation generated when you open a PR and can be accessed using the Github UI.
+
+1. click the checkmark next to your commit and it will expand to have multiple options
```
[the remark this answers is not in the document]
Comment (astrojuanlu) [reply docs/development/docs.rst]: Let's continue the discussion over #8108 👍🏽

**Review by astrojuanlu — APPROVED** [review]
I think this is definitely an improvement, and I'm happy that it sparked a couple of useful conversations (#8090, #8108). Thanks a lot @cocobennett!

File: docs/development/docs.rst [hunk 14]
```diff
@@ -5,32 +5,77 @@ As one might expect,
...
+
+      git clone --recurse-submodules https://github.com/readthedocs/readthedocs.org/
+
+#. install documentation requirements
+
+   .. prompt:: bash
+
+      cd readthedocs.org
+      pip install -r requirements/testing.txt
+      pip install -r requirements/docs.txt
```
Comment (humitos) [inline docs/development/docs.rst:44]: Shouldn't we walk the user through the steps to create a virtualenv for this?

File: docs/development/docs.rst [hunk 15]
```diff
@@ -5,32 +5,77 @@ As one might expect,
...
+   .. prompt:: bash
+
+      git clone --recurse-submodules https://github.com/readthedocs/readthedocs.org/
+
+#. install documentation requirements
+
+   .. prompt:: bash
+
+      cd readthedocs.org
+      pip install -r requirements/testing.txt
```
Comment (humitos) [inline docs/development/docs.rst:43]: Do we need `requirements/testing.txt` to work on the docs?

Comment (astrojuanlu) [reply docs/development/docs.rst:43]: At the moment they're needed, yes. Otherwise some dependencies are missing.

Comment (astrojuanlu) [reply docs/development/docs.rst:43]: (It's up to debate whether we should organize our dependencies in a better way, or improve the naming of the requirements files, see https://github.com/readthedocs/readthedocs.org/pull/7037#issuecomment-644960726)

**Pull request merged**